server/util/index.ts

```diff
@@ -14,3 +17,53 @@ export function formatAddress(address: string): string {
   }
   return addressString.formattedAddress;
 }
+
+function validateToken(
+  req: Request,
+  res: Response,
+  callback: (payload?: JWTPayload) => void,
+) {
+  const { authorization } = req.headers;
+  if (authorization) {
+    const [bearer, token] = authorization.split(' ');
+    if (bearer === 'Bearer') {
+      jwt.verify(token, process.env.JWT_SECRET!, (err, payload) => {
+        if (err) {
+          res.send({ err });
+        } else {
+          callback(payload as JWTPayload);
+        }
+      });
+    }
+  } else {
+    res.send({ err: 'No token provided' });
+  }
+}
+
+function isUserValid(userType: UserType, authLevel: UserType) {
+  const priority: { [type in UserType]: number } = {
+    User: 0,
+    Rider: 1,
+    Driver: 1,
+    Dispatcher: 2,
+  };
```

Comment:
Could define this outside of the function if you really wanted to. That way there is only ever 1 copy of this Map.